File: 0216-combination-sum-iii/0216-combination-sum-iii.cpp
class Solution {
private:
    void findCombination(int index,vector<vector<int>> &result,vector<int>temp, int k, int target,vector<int> nums){
        if(target == 0 && temp.size() == k){
            result.push_back(temp);
            return;
        }
        for(int i=index; i<nums.size(); i++){
             if (nums[i] > target) {  // If the number exceeds the target, no need to continue
                break;
                 }
                temp.push_back(nums[i]);
                findCombination(i+1, result, temp, k, target-nums[i],nums);
                temp.pop_back();
            
        }
    }
public:
    vector<vector<int>> combinationSum3(int k, int n) {
      vector<int> nums = {1,2,3,4,5,6,7,8,9};
      vector<vector<int>> result;
      vector<int> temp;
      findCombination(0,result,temp,k,n,nums);
      return result;
    }
};

File: 0216-combination-sum-iii/0216-combination-sum-iii.java
class Solution {
    private void findCombination(int index,int k, int target, List<Integer> temp, List<List<Integer>> result){
        if(temp.size() == k && target == 0){
            result.add(new ArrayList<>(temp));
            return;
        }

        for(int i=index ; i< 9; i++){
            temp.add(i+1);
            findCombination(i+1,k,target-(i+1),temp,result);
            temp.remove(temp.size()-1);
        }
    }
    public List<List<Integer>> combinationSum3(int k, int n) {
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        findCombination(0,k,n,temp,result);
        return result;
    }
}